cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/assembly/KafkaConnectApiImpl.java

```diff
@@ -473,9 +475,11 @@ void rescheduleOrComplete(Long tid) {
             }
             return k1.compareTo(k2);
         });
-        updateLoggers.putAll(fetchedLoggers);
+        Map<String, String> desiredMap = new OrderedProperties().addStringPairs(Util.expandVars(desiredLogging)).asMap();
+
+        fetchedLoggers.entrySet().forEach(fetchedLogger -> updateLoggers.put(fetchedLogger.getKey(), getLevel(fetchedLogger.getKey(), desiredMap)));
```

Comment:
You could use `fetchedLoggers.forEach`, rather than going via `entrySet()`, but I still don't really like that. To me this is clearer, even if it's longer, because `updateLoggers` is populated once, rather than in an implicit loop:

```java
        updateLoggers.putAll(fetchedLoggers.keySet().stream().collect(Collectors.toMap(
                Function.identity(),
                key -> getEffectiveLevel(key, desiredMap))));
```

WDYT?